### ISSUE TITLE
[a11y]: Data table may be missing some id elements that may affect accessibility. 

### ISSUE THREAD

**Issue by xmannyxfreshx1** [carbon-design-system/carbon#11998]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

v7.56.0

### React version

v17.0.2

### Automated testing tool and ruleset

IBM equal access accessibility checker - latest

### Description

IBM equal access accessibility checker is flagging a table component as needing an `id` element.
<img width="1728" alt="Screen Shot 2022-08-12 at 4 15 34 PM" src="https://user-images.githubusercontent.com/13371221/186169636-26a15739-e8e6-4660-9d0f-865adad63a21.png">
 

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-datatable-expansion--default

### Steps to reproduce

Run a scan (IBM equal access accessibility checker plugin) on the page listed above to see the violation.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [ ] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by jnm2377** [comment]
Hey @xmannyxfreshx1 thanks for taking the time to open this issue. Looks like this is related to the changes made here: https://github.com/carbon-design-system/carbon/pull/10214 https://github.com/carbon-design-system/carbon/pull/10547 https://github.com/carbon-design-system/carbon/pull/11217 but we forgot to update our website example. I think rather than updating the story to include an `id`, we should set the default `id` prop to equal `expand` since this is the [default name](https://github.com/carbon-design-system/carbon/blob/ec338ee5b106b78aaafbd5d0121d8dae3a6434bd/packages/react/src/components/DataTable/TableExpandRow.js#L96) we give expand headers, and users can change them if they want a different id/name. This will also ensure users don't have an a11y violation by default.
